File: sub-module-web/src/main/java/org/github/web/controller/SsoClientController.java
package org.github.web.controller;

import org.github.core.spring.restful.json.JSONDataReturn;
import org.github.core.spring.restful.json.JSONReturn;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import static cn.dev33.satoken.sso.SaSsoProcessor.instance;
import static cn.dev33.satoken.sso.SaSsoUtil.buildServerAuthUrl;
import static cn.dev33.satoken.stp.StpUtil.getTokenValue;
import static cn.dev33.satoken.stp.StpUtil.isLogin;
import static cn.dev33.satoken.stp.StpUtil.login;
import static org.springframework.http.HttpStatus.UNAUTHORIZED;

@RestController
public class SsoClientController {
  @RequestMapping("/sso/doLoginByTicket")
  public JSONReturn doLoginByTicket(String ticket) {
    Object loginId = instance.checkTicket(ticket, "/sso/doLoginByTicket");
    if (loginId == null) return JSONReturn.of(UNAUTHORIZED, "无效ticket：" + ticket);
    login(loginId);
    String token = getTokenValue();
    return JSONDataReturn.of(token);
  }

  @RequestMapping("/sso/getSsoAuthUrl")
  public JSONReturn getSsoAuthUrl(String clientLoginUrl) {
    String serverAuthUrl = buildServerAuthUrl(clientLoginUrl, "");
    return JSONDataReturn.of(serverAuthUrl);
  }

  @RequestMapping("/sso/hasLogin")
  public Object hasLogin() {
    boolean flag = isLogin();
    return JSONDataReturn.of(flag);
  }

  @RequestMapping("/sso/*")
  public Object ssoRequest() {
    return instance.clientDister();
  }
}

File: sub-system-sso/src/main/java/org/github/sso/controller/SsoClientController.kt
package org.github.sso.controller

import cn.dev33.satoken.sso.SaSsoProcessor.instance
import cn.dev33.satoken.sso.SaSsoUtil.buildServerAuthUrl
import cn.dev33.satoken.stp.StpUtil.getTokenValue
import cn.dev33.satoken.stp.StpUtil.isLogin
import cn.dev33.satoken.stp.StpUtil.login
import org.github.core.DEVICE_TYPE_PC
import org.github.core.exception.ClientException
import org.github.core.spring.restful.json.JSONData
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class SsoClientController {
  @RequestMapping("/sso/doLoginByTicket")
  fun doLoginByTicket(ticket: String): JSONData<String> {
    val loginId = instance.checkTicket(ticket, "/sso/doLoginByTicket") ?: throw ClientException("无效ticket：$ticket")
    login(loginId, DEVICE_TYPE_PC)
    val token = getTokenValue()!!
    return JSONData.of(token)
  }

  @RequestMapping("/sso/getSsoAuthUrl")
  fun getSsoAuthUrl(clientLoginUrl: String): JSONData<String> {
    val serverAuthUrl = buildServerAuthUrl(clientLoginUrl, "")!!
    return JSONData.of(serverAuthUrl)
  }

  @RequestMapping("/sso/hasLogin")
  fun hasLogin(): JSONData<Boolean> = JSONData.of(isLogin())

  @RequestMapping("/sso/*")
  fun ssoRequest(): Any = instance.clientDister()
}
